typed-racket-lib/typed-racket/types/abbrev.rkt

```diff
@@ -300,3 +300,11 @@
            [_ rep])))
      (values new-type props)]))
 
+
+
+(define (-list-set s) (Un (-lst s) (-set s)))
+
```

Comment:
Would `-list-or-set` be clearer for future maintainers?

Reply:
Yes, I think so.